components/movementsensor/adxl345/adxl345.go

```diff
@@ -189,10 +266,113 @@ func NewAdxl345(
 			}
 		}
 	})
-
+	sensor.interruptsFound = make(map[string]interruptInfo)
+	sensor.readInterrupts(sensor.cancelContext)
+	err = sensor.configureInterruptRegisters(ctx)
+	if err != nil {
+		sensor.cancelFunc()
+		return nil, err
+	}
+	if sensor.echoInterrupt1 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt1)
+	}
+	if sensor.echoInterrupt2 != nil {
+		sensor.startInterruptPolling(sensor.echoInterrupt2)
+	}
 	return sensor, nil
 }
 
+func (adxl *adxl345) startInterruptPolling(interrupt board.DigitalInterrupt) {
+	utils.PanicCapturingGo(func() {
+		interrupt.AddCallback(adxl.ticksChan)
+		defer interrupt.RemoveCallback(adxl.ticksChan)
+
+		for {
+			select {
+			case <-adxl.cancelContext.Done():
+				return
+			case tick := <-adxl.ticksChan:
+				if tick.High {
+					func() {
+						adxl.mu.Lock()
+						defer adxl.mu.Unlock()
+						adxl.readInterrupts(adxl.cancelContext)
+					}()
+				}
+			}
+		}
+	})
+}
+
+func getInterruptConfigurations(int1, int2 InterruptPinAttrConfig) map[byte]byte {
+	var intEnabled byte
+	var intMap byte
+
+	if int1.SingleTap {
+		intEnabled += interruptBitPosition[SingleTap]
+	} else if int2.SingleTap {
+		intEnabled += interruptBitPosition[SingleTap]
+		intMap += interruptBitPosition[SingleTap]
+	}
+
+	if int1.FreeFall {
+		intEnabled += interruptBitPosition[FreeFall]
+	} else if int2.FreeFall {
+		intEnabled += interruptBitPosition[FreeFall]
+		intMap += interruptBitPosition[FreeFall]
+	}
+	return map[byte]byte{IntEnableAddr: intEnabled, IntMapAddr: intMap}
+}
+
+// This returns a map from register addresses to data which should be written to that register to configure the interrupt pin.
+func getSingleTapRegisterValuesFromInterruptPin(interruptPin InterruptPinAttrConfig, registerValues map[byte]byte) map[byte]byte {
+	singleTapConfigs := interruptPin.TapAttrConfig
+	if singleTapConfigs == nil {
+		return registerValues
+	}
+
+	tapAxesSpecified := singleTapConfigs.ExcludeTapX || singleTapConfigs.ExcludeTapY || singleTapConfigs.ExcludeTapZ
```

Comment:
tapAcesSpecificed needs to be negated

Reply:
This is just to check if the user wants to exclude any of the axes. Otherwise they will all be included because these default to false. The actual logic for setting the register in the `getAxes` function.

Reply:
Although this is unnecessary logic and leftover from when it was more complicated so I will just remove it